### PR TITLE
Change ingress api doc to reflect the correct path validation

### LINKED ISSUE #101594: networking.k8s.io/v1 fails with default path
#### What happened:
When I tried to use the default path I got `Ingress.extensions "nginx" is invalid: spec.rules[0].http.paths[0].path: Invalid value: "": must be an absolute path`.

#### What you expected to happen:
No error. According to https://docs.openshift.com/container-platform/4.6/rest_api/network_apis/ingress-networking-k8s-io-v1.html the path is not needed. "When unspecified, all paths from incoming requests are matched."

#### How to reproduce it (as minimally and precisely as possible):
Create a ingress like so:
```yml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: nginx
  namespace: {{ .Release.Namespace }}
  annotations:
    kubernetes.io/ingress.class: "alb"
    alb.ingress.kubernetes.io/scheme: "internet-facing"
    alb.ingress.kubernetes.io/target-type: "ip"
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP":80,"HTTPS": 443}]'
    alb.ingress.kubernetes.io/certificate-arn: {{ .Values.arn_cert }}
    alb.ingress.kubernetes.io/healthcheck-path: "/"
    alb.ingress.kubernetes.io/success-codes: "200,404"
spec:
  rules:
    - host: "*.foo.com"
      http:
        paths:
          - backend:
              service:
                name: foo
                port:
                  name: http
            pathType: "Exact"
```

#### Environment:
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-10T12:53:33Z", GoVersion:"go1.16.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.8-eks-96780e", GitCommit:"96780e1b30acbf0a52c38b6030d7853e575bcdf3", GitTreeState:"clean", BuildDate:"2021-03-10T21:32:29Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
```

- Cloud provider or hardware configuration: AWS
- OS (e.g: `cat /etc/os-release`): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: brew
- Network plugin and version (if this is a network-related bug): N/A
- Others:

### PR DESCRIPTION
**Pull request by rikatz**
Signed-off-by: Ricardo Pchevuzinske Katz <ricardo.katz@gmail.com>

/kind cleanup
/kind documentation

#### What this PR does / why we need it:
The API documentation does not reflect the current ingress validation:

https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/pkg/apis/networking/validation/validation.go#L397-L399

While it states that if "path" is empty it catches all, this is a specific behavior for PathType=ImplementationSpecific.

The idea here is to be more explicit that 'path' is actually required if you are using pathType of 'exact' or 'prefix'.

#### Which issue(s) this PR fixes:
Fixes #101594

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/sig network

### CODE REVIEW

**Comment by rikatz** [comment]
/cc @robscott 

Rob, if you can take a look into this, so we can avoid some more confusions :)

Thanks!

File: pkg/apis/networking/types.go [hunk 1]
```diff
@@ -492,8 +492,8 @@ const (
 type HTTPIngressPath struct {
 	// Path is matched against the path of an incoming request. Currently it can
 	// contain characters disallowed from the conventional "path" part of a URL
-	// as defined by RFC 3986. Paths must begin with a '/'. When unspecified,
-	// all paths from incoming requests are matched.
+	// as defined by RFC 3986. Paths must begin with a '/' and must be present
+	// when using PathType with value "Exact" or "Prefix".
```
Comment (robscott) [inline pkg/apis/networking/types.go:496]: Tiny nit:
```suggestion
	// when using "Exact" or "Prefix" as PathType.
```

**Review by robscott — COMMENTED** [review]
Thanks for fixing this! This wording is _much_ clearer.

**Comment by robscott** [comment]
Thanks!

/lgtm
/retest
/assign @thockin

**Pull request merged**